tools/nni_trial_tool/trial.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import ctypes
+import os
+import shlex
+import tarfile
+import time
+from datetime import datetime
+from subprocess import Popen
+
+import psutil
+
+from .log_utils import LogType, RemoteLogger, StdOutputType, nni_log
+from .base_channel import CommandType
+
+trial_output_path_name = ".nni"
+
+
+class Trial:
+    def __init__(self, args, data):
+        self.process = None
+        self.data = data
+        self.args = args
+        self.command_channel = args.command_channel
+        self.trial_syslogger_stdout = None
+
+        global NNI_TRIAL_JOB_ID
+        self.id = data["trialId"]
+        if self.id is None:
+            raise Exception("trial_id is not found in %s" % data)
+        os.environ['NNI_TRIAL_JOB_ID'] = self.id
+        NNI_TRIAL_JOB_ID = self.id
+
+        # for multiple nodes. If it's None, it means single node.
+        self.node_id = args.node_id
+        if self.node_id is None:
+            self.name = self.id
+        else:
+            self.name = "%s_%s" % (self.id, self.node_id)
+
+    def run(self):
+        # redirect trial's stdout and stderr to syslog
+        self.trial_syslogger_stdout = RemoteLogger(self.args.nnimanager_ip, self.args.nnimanager_port, 'trial', StdOutputType.Stdout,
+                                                   self.args.log_collection, self.id)
+
+        nni_log(LogType.Info, "%s: start to run trial" % self.name)
+
+        trial_working_dir = os.path.realpath(os.path.join(os.curdir, "..", "..", "trials", self.id))
```

Comment:
a little confused about the folder structure. trial_work_dir is in top level?

Reply:
trial_working_dir should be root dir of a trial, it has code, .nni folder inside.